webboot/webboot.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/u-root/u-root/pkg/dhclient"
+	"github.com/vishvananda/netlink"
+)
+
+var (
+	ifName = flag.String("InterfaceName", "^e.*","Name of the interface" ) //turn into a flag.. 
+	timeout = flag.Int("timeout", 15, "Lease timeout in seconds")
+	retry = flag.Int("retry", 5, "Max number of attempts for DHCP clients to send requests. -1 means infinity")
+	verbose = flag.Bool("verbose", false, "Verbose output")
+	ipv4 = flag.Bool("ipv4", true, "use IPV4")
+	ipv6 = flag.Bool("ipv6", true, "use IPV6")
+)
+
+func dhclientNew(ifName string, timeout int, retry int, verbose bool, ipv4 bool, ipv6 bool) {
+	ifRE := regexp.MustCompilePOSIX(ifName)
+
+	ifnames, err := netlink.LinkList()
+	if err != nil {
+		log.Fatalf("Can't get list of link names: %v", err)
+	}
+
+	var filteredIfs []netlink.Link
+	for _, iface := range ifnames {
+		if ifRE.MatchString(iface.Attrs().Name) {
+			filteredIfs = append(filteredIfs, iface)
+		}
+	}
+
+	if len(filteredIfs) == 0 {
+		log.Fatalf("No interfaces match %s", ifName)
+	}
+
+	channel := make(chan bool)
+	go configureAlls(filteredIfs,channel)
+	x := <-channel
+}
+
+func configureAlls(ifs []netlink.Link, channel chan bool) {
```

Comment:
make this a chan of errors, then you can push nil errors to say "it worked" and errors are useful

Reply:
Good idea :)!